Web/3DObject/rollup.config.js

```diff
@@ -1,13 +1,37 @@
 import typescript from "@rollup/plugin-typescript";
+import fs from "fs";
 
 export default args => {
     const result = args.configDefaultConfig;
+    result.treeshake = false;
     result.forEach(config => {
+        const external = [/^@babylonjs\/core($|\/)/];
+        config.external = [...config.external, ...external];
         config.plugins.push(
             typescript({
                 include: ["../../Shared/ComponentParent/**/*.ts+(|x)", "./**/*.ts+(|x)"]
-            })
+            }),
+            {
+                writeBundle(bundle) {
+                    for (const [fileName, chunkOrAsset] of Object.entries(bundle)) {
+                        if (
+                            chunkOrAsset !== undefined &&
+                            (chunkOrAsset.toString().endsWith(".js") || chunkOrAsset.toString().endsWith(".mjs")) &&
+                            fileName === "file"
+                        ) {
+                            const data = fs.readFileSync(chunkOrAsset, { encoding: "utf8" });
+                            const regex = /@babylonjs\/core\/.*?\.js/g;
+                            fs.writeFileSync(chunkOrAsset, data.replaceAll(regex, "../../../shared/babylonjscore"));
+                        }
+                    }
+                }
+            }
```

Comment:
1. Why `writeBundle` plugin is used only in this config?
2. Why we need regex replace? I think rollup can take care of this.

Reply:
The problem I had was that since it's importing two loaders from babylon it would create new imports that I didn't get changed in a previous step. So it would change all the imports to my bundled file, and after that resolve the loader imports, and I'd still have references to babylon/core instead of the bundled file. 
Since writeBundle is the very last step, it works and catches everything